src/LanguageServerProtocol/LanguageServerProtocol.fs

```diff
@@ -1879,13 +1904,15 @@ module JsonRpc =
         [<JsonProperty("jsonrpc")>] Version: string
         Id: int option
         Error: Error option
+        [<JsonProperty(NullValueHandling=NullValueHandling.Include)>]
         Result: JToken option
     }
     with
-        static member Success(id: int option, result: JToken option) =
-            { Version = "2.0"; Id = id; Result = result; Error = None }
-        static member Failure(id: int option, error: Error) =
-            { Version = "2.0"; Id = id; Result = None; Error = Some error }
+        member x.ShouldSerializeResult() = x.Error.IsNone
```

Comment:
can this be just a precomputed property? `member x.ShouldSerializeResult = x.Error.IsNone` should only compute the value one, IIRC.

Reply:
https://www.newtonsoft.com/json/help/html/ConditionalProperties.htm
This is a magic Json.NET naming convention. It probably needs to be a method I think? Added a clarifying comment either way.